pkg/platform/authentication/auth.go

```diff
@@ -235,23 +235,36 @@ func (s *Auth) Logout() {
 
 // Client will return an API client that has authentication set up
 func (s *Auth) Client() *mono_client.Mono {
+	client, err := s.ClientSafe()
+	if err != nil {
+		logging.Error("Trying to get the Client while not authenticated")
+		fmt.Fprint(os.Stderr, locale.T("err_api_not_authenticated"))
+		exit(1)
```

Comment:
Big no! If we need to handle this error let's do so properly. So let's remove ClientSafe and have Client() return an error.

Exit handling is a crucial part of the new command structure, working around it in what is meant to be refactored code is not an option.

Reply:
I provided ClientSafe to make the changes to the secrets command, but modifying Client itself will mean many cascading changes. I do think ClientSafe should not be necessary, but I also thought sprawling changes to remove it would make most sense in another story. Which should have been filed already, sorry. I'm happy to take direction from here otherwise.